tests/trinity/conftest.py

```diff
@@ -64,11 +95,16 @@ def p2p_server(monkeypatch, jsonrpc_ipc_pipe_path):
     return Server(None, None, None, None, None, None, None)
 
 
+@pytest.fixture
+def chain_with_block_validation(base_db, genesis_state):
+    return _chain_with_block_validation(base_db, genesis_state, TestAsyncChain)
```

Comment:
maybe use the `evm.tools.builder` utils for this (to reduce shared reliance on the `chain_with_block_validation` fixture).  I'm of the opinion that the broader we use that fixture the more likely we are to end up in a situation where we have a tangled mess of dependencies that all need slightly different things from it.

Reply:
Just to be clear on the problem I tried to solve here: Certain tests in `tests/trinity` are already depending on the `chain_with_block_validation` fixture that is in `tests/conftest`. That however is tight to a chain that doesn't have the `async` functions and I can't import from Trinity there.

So, I created another `chain_with_block_validation` fixture in `trinity/conftest` to break the lookup one level earlier and pass the `TestAsyncChain`.